.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,75 @@
+name: Test suite
+
+on:
+  push:
+  pull_request:
+  schedule:
+  - cron: 1 0 * * *  # Run daily at 0:01 UTC
+
+jobs:
+  tests:
+    name: 👷
+    runs-on: ${{ matrix.os }}
+    strategy:
```

Comment:
It's important to note that by default any job failure will cause all other jobs in the workflow to be canceled. If you don't want that, apply this patch:
```suggestion
    strategy:
      fail-fast: false
```

Reply:
I've found this useful, especially if there's a network-related timeout in one job.